src/js/background.ts

```diff
@@ -135,67 +162,108 @@ function handleMessages(
+        } else {
+          sendResponse({
+            valid: false,
+            hash: jsHash,
+            reason:
+              'Error: hash does not match ' +
+              message.origin +
+              ', ' +
+              message.version +
+              ', unmatched JS is ' +
+              message.rawjs,
+          });
+        }
+      });
+
+      // Indicates that the message will send an async response.
+      return true;
     }
-    const manifestObj = origin.get(message.version);
-    const manifest = manifestObj && manifestObj.leaves;
-    if (!manifest) {
-      sendResponse({valid: false, reason: 'no matching manifest'});
+
+    case MESSAGE_TYPE.UPDATE_STATE: {
+      updateContentScriptState(validSender, message.state, message.origin);
+      sendResponse({success: true});
       return;
     }
 
-    // fetch the src
-    const encoder = new TextEncoder();
-    const encodedJS = encoder.encode(message.rawjs);
-    // hash the src
-    crypto.subtle.digest('SHA-256', encodedJS).then(jsHashBuffer => {
-      const jsHashArray = Array.from(new Uint8Array(jsHashBuffer));
-      const jsHash = jsHashArray
-        .map(b => b.toString(16).padStart(2, '0'))
-        .join('');
-
-      if (manifestObj.leaves.includes(jsHash)) {
-        sendResponse({valid: true});
-      } else {
+    case MESSAGE_TYPE.CONTENT_SCRIPT_START: {
+      recordContentScriptStart(validSender, message.origin);
+
+      if (!message.checkCSPHeaders) {
         sendResponse({
-          valid: false,
-          hash: jsHash,
-          reason:
-            'Error: hash does not match ' +
-            message.origin +
-            ', ' +
-            message.version +
-            ', unmatched JS is ' +
-            message.rawjs,
+          success: true,
         });
+
+        return;
+      }
+
+      const cspHeadersByTab = CSP_HEADERS.get(validSender.tab.id);
+      if (!cspHeadersByTab) {
+        // This may indicate that setupCSPListener has failed to fire.
+        invalidateContentScriptStateAndThrow(
+          validSender,
+          message.origin,
+          'No CSP headers have been stored for this tab ID',
+        );
       }
-    });
-    return true;
-  } else if (message.type === MESSAGE_TYPE.UPDATE_STATE) {
-    updateContentScriptState(sender, message.state, message.origin);
-    sendResponse({success: true});
-  } else if (message.type === MESSAGE_TYPE.CONTENT_SCRIPT_START) {
-    recordContentScriptStart(sender, message.origin);
-    sendResponse({
-      success: true,
-      cspHeaders: CSP_HEADERS?.get(sender.tab.id)?.get(sender.frameId),
-      cspReportHeaders: CSP_REPORT_HEADERS?.get(sender.tab.id)?.get(
-        sender.frameId,
-      ),
-    });
-  } else if (message.type === MESSAGE_TYPE.UPDATED_CACHED_SCRIPT_URLS) {
-    if (!CACHED_SCRIPTS_URLS.has(sender.tab.id)) {
-      CACHED_SCRIPTS_URLS.set(sender.tab.id, new Set());
+
+      sendResponse({
+        success: true,
+        // eslint-disable-next-line
```

Comment:
What are suppressing here?

Reply:
Oh, I think I was going back and forth a bit on how to handle this and forgot to remove this. Good catch!